normalize/diff.py

```diff
@@ -660,6 +664,20 @@ def compare_collection_iter(propval_a, propval_b, fs_a=None, fs_b=None,
             rev_key[(pk, seen[pk])] = k
             seen[pk] += 1
 
+    if options.recurse:
+        # we can be sure that both records have these keys
+        shared_keys = set(rev_key.values())
```

Comment:
This `rev_key` here is **just** the keys on the right side; it's a free variable from the last iteration of the loop in the block before it.  I'm guessing that's not what you meant.